vm/src/obj/objbool.rs

```diff
@@ -71,6 +77,72 @@ fn bool_repr(vm: &VirtualMachine, args: PyFuncArgs) -> Result<PyObjectRef, PyObj
     Ok(vm.new_str(s))
 }
 
+fn do_bool_or(vm: &VirtualMachine, lhs: &PyObjectRef, rhs: &PyObjectRef) -> PyResult {
+    if objtype::isinstance(lhs, &vm.ctx.bool_type())
+        && objtype::isinstance(rhs, &vm.ctx.bool_type())
+    {
+        let lhs = get_value(lhs);
+        let rhs = get_value(rhs);
+        (lhs || rhs).into_pyobject(vm)
+    } else {
+        Ok(lhs.payload::<PyInt>().unwrap().or(rhs.clone(), vm))
+    }
+}
+
+fn bool_or(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(vm, args, required = [(lhs, None), (rhs, None)]);
+    do_bool_or(vm, lhs, rhs)
+}
+
+fn bool_ror(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(vm, args, required = [(rhs, None), (lhs, None)]);
+    do_bool_or(vm, lhs, rhs)
+}
+
+fn do_bool_and(vm: &VirtualMachine, lhs: &PyObjectRef, rhs: &PyObjectRef) -> PyResult {
+    if objtype::isinstance(lhs, &vm.ctx.bool_type())
+        && objtype::isinstance(rhs, &vm.ctx.bool_type())
+    {
+        let lhs = get_value(lhs);
+        let rhs = get_value(rhs);
+        (lhs && rhs).into_pyobject(vm)
+    } else {
+        Ok(lhs.payload::<PyInt>().unwrap().and(rhs.clone(), vm))
+    }
+}
+
+fn bool_and(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(vm, args, required = [(lhs, None), (rhs, None)]);
+    do_bool_and(vm, lhs, rhs)
+}
+
+fn bool_rand(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(vm, args, required = [(rhs, None), (lhs, None)]);
+    do_bool_and(vm, lhs, rhs)
+}
+
+fn do_bool_xor(vm: &VirtualMachine, lhs: &PyObjectRef, rhs: &PyObjectRef) -> PyResult {
+    if objtype::isinstance(lhs, &vm.ctx.bool_type())
+        && objtype::isinstance(rhs, &vm.ctx.bool_type())
+    {
+        let lhs = get_value(lhs);
+        let rhs = get_value(rhs);
+        (lhs ^ rhs).into_pyobject(vm)
+    } else {
+        Ok(lhs.payload::<PyInt>().unwrap().and(rhs.clone(), vm))
```

Comment:
This should probably use `.xor` instead of `.and`?

Reply:
oops